spec/classes/qpid_client_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe 'katello::qpid_client' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+
+      if facts[:operatingsystemmajrelease] == '7'
```

Comment:
A better practice is to not shadow variables. You can also add conditionals to a context.
```suggestion
  on_supported_os.each do |os, facts|
    context "on #{os}", if: os_facts[:operatingsystemmajrelease] == '7' do
      let(:facts) { os_facts }

```

Reply:
This check is no longer there!